src/Calculator/BcMathCalculator.php

```diff
@@ -31,13 +32,17 @@ public static function compare(string $a, string $b): int
     /** @psalm-pure */
     public static function add(string $amount, string $addend): string
     {
-        return bcadd($amount, $addend, self::SCALE);
+        $scale = str_contains($amount, '.') ? self::SCALE : 0;
```

Comment:
This is checking `$amount`, but not `$addend`

Reply:
I'd really just leave these methods alone :P

Reply:
In fact, why not patch bcmath itself to ignore the scale directly, when there's no decimal stuff going on? :P

Reply:
Not a breakage, if there's no behavioral change.

If there's noticeable performance overhead, a large chunk of the PHP community could benefit from it.

Reply:
I need beauviour change for the performance improvement in the MoneyPHP ctor. 

See https://github.com/moneyphp/money/pull/747#issuecomment-1498907505

Reply:
Sorry, lemme rephrase: I was saying that patching https://github.com/php/php-src/ is not a BC break if there's no behavioral change.

Reply:
I think we talk past each other.

`bcadd('5', '15', 0)` and `bcadd('5', '15', 14)` are already comparable performance wise.

What is different is the returned string value:

```php
bcadd('5', '15', 0); // string(2) "20"
bcadd('5', '15', 14); // string(2) "20.00000000000000"
```

I'd need to change in the php-core. 

```diff
- bcadd('5', '15', 14); // string(2) "20.00000000000000"
+ bcadd('5', '15', 14); // string(2) "20"
```

I consider this a major BC break.


The actual performance improvement comes from passing the calculatores return value into `Money::__construct`


```php
return new Money("20", $currenty); // fast
return new Money("20.00000000000000", $currenty); // fast
```

This is due to additional number parsing if the passed string is valid for `FILTER_VALIDATE_INT `: https://github.com/moneyphp/money/blob/8953b873725c465bc55eee12a1c91b923e0d63c7/src/Money.php#L80

That given maybe it's would be a better fix to improve that validation for numeric strings containing int values.

So a fast coercion from `20.00000000000000` to `20`.

I hope that clears things up. Sorry for splitting my PR into two, that whas not helpful to preserve context.